### PR TITLE
Add feature to support customized config files to the cluster

### LINKED ISSUE #73: [FEATURE]Ability to provide customized security configurations to the cluster
### Is your feature request related to a problem?
The current opensearch-cluster-cdk configuration in terms of security is either `default` security config or no security at all. 
It would be helpful to be able to provide custom security configuration to the cluster. For example, only specific role mapping, allow anonymous read, etc

### What solution would you like?
Ability to provide customized security configurations to the cluster

### What alternatives have you considered?
Manually logging into the cluster, changing config followed by cluster restart

### Do you have any additional context?
_Add any other context or screenshots about the feature request here._

### PR DESCRIPTION
**Pull request by gaiksaya**
### Description
The `customConfigFiles` feature allows the users to overwrite or add custom config files to OpenSearch and OpenSearch Dashboards. This will allow to provide any custom configurations including security making the cluster generic for any use cases.

### Issues Resolved
resolves #73 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -68,6 +68,7 @@ In order to deploy both the stacks the user needs to provide a set of required a
...
 | customRoleArn          | Optional    | string  | User provided IAM role arn to be used as ec2 instance profile. `-c customRoleArn=arn:aws:iam::<AWS_ACCOUNT_ID>:role/<ROLE_NAME>`                                                                                                                                                                 |
+| customConfigFiles          | Optional    | string  | You can provide an entire config file to be overwritten or added to OpenSearch and OpenSearch Dashboards. Pass string in the form of JSON with key as local path to the config file to read from and value as file on the server to overwrite/add. Note that the values in the JSON needs to have prefix of `opensearch` or `opensearch-dashboards`. Example: `-c customConfigFiles='{"opensearch-config/config.yml": "opensearch/config/opensearch-security/config.yml", "opensearch-config/role_mapping.yml":"opensearch/config/opensearch-security/roles_mapping.yml", "/roles.yml": "opensearch/config/opensearch-security/roles.yml"}'`                                                                                                                                                                 |
```
Comment (rishabh6788) [inline README.md:71]: Nit: Formatting change.

File: lib/infra/infra-stack.ts [hunk 2]
```diff
@@ -585,7 +586,25 @@ export class InfraStack extends Stack {
...
     }
 
-    // // Startinng OpenSearch based on whether the distribution type is min or bundle
+    if (props.customConfigFiles !== 'undefined') {
+      try {
+        // @ts-ignore
+        const jsonObj = JSON.parse(props.customConfigFiles);
+        Object.keys(jsonObj).forEach((localFileName) => {
+          const getConfig = load(readFileSync(localFileName, 'utf-8'));
+          const remoteConfigLocation = jsonObj[localFileName];
```
Comment (rishabh6788) [inline lib/infra/infra-stack.ts:595]: I am assuming this provides the value, i.e, the actual path of the file in opensearch or opensearch-dashboards directory on the ec2 host?

Comment (gaiksaya) [reply lib/infra/infra-stack.ts:595]: Yes! root directory remains the same which is `/home/ec2-user`, the value of json needs to be `opensearch/config/*` etc.

File: README.md (hunk 1 above)
Comment (gaiksaya) [reply README.md]: Fixed! Thanks!

File: lib/infra/infra-stack.ts (hunk 2 above)
Comment (rishabh6788) [reply lib/infra/infra-stack.ts:595]: Also, for local file name, the user has to provide the absolute path or does the code reads using relative-path?

Comment (gaiksaya) [reply lib/infra/infra-stack.ts:595]: Both works as per my testing. Relative as well as absolute path.

**Review by rishabh6788 — APPROVED** [review]

**Pull request merged**